SU2_CFD/src/objectives/turbomachinery/CTurbomachineryPerformance.cpp

```diff
@@ -0,0 +1,177 @@
+/*!
+ * \file CTurbomachineryPerformance.cpp
+ * \brief Source of the Turbomachinery Performance class.
+ * \author S. Vitale
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../include/objectives/turbomachinery/CTurbomachineryPerformance.hpp"
+
+CTurbomachineryPrimitiveState::CTurbomachineryPrimitiveState(){
+  Density = Pressure = TangVelocity = 0.0;
+}
+CTurbomachineryPrimitiveState::CTurbomachineryPrimitiveState(su2double density, su2double pressure, su2double *velocity, unsigned short nDim, su2double tangVel) : 
+                              Density(density), Pressure(pressure), TangVelocity(tangVel){
+  Velocity.assign(velocity, velocity + nDim );
+}
+
+CTurbomachineryPrimitiveState::~CTurbomachineryPrimitiveState(){}
+
+CTurbomachineryCombinedPrimitiveStates::CTurbomachineryCombinedPrimitiveStates(const CTurbomachineryPrimitiveState& inletPrimitiveState, 
+                                        const CTurbomachineryPrimitiveState&  outletPrimitiveState) : InletPrimitiveState(inletPrimitiveState), OutletPrimitiveState(outletPrimitiveState){}
+
+CTurbomachineryCombinedPrimitiveStates::~CTurbomachineryCombinedPrimitiveStates(){}
+
+CTurbomachineryState::CTurbomachineryState(){
+  Density = Pressure = Entropy = Enthalpy = Temperature = TotalTemperature = TotalPressure = TotalEnthalpy = 0.0;
+  AbsFlowAngle = FlowAngle = MassFlow = Rothalpy = TotalRelPressure = 0.0;
+  Area = Radius = 0.0;
```

Comment:
In c++11 this type of default value initialization can be done in the declaration of the class members directly, which saves a lot of repetition.